home/tests/test_serializers.py

```diff
@@ -0,0 +1,285 @@
+import pytest
+
+from home.models import ContentPage, HomePage
+from home.serializers import (
+    format_whatsapp_message,
+    has_next_message,
+    has_previous_message,
+)
+
+from .page_builder import (
+    MBlk,
+    MBody,
+    PageBuilder,
+    SBlk,
+    SBody,
+    UBlk,
+    UBody,
+    VarMsg,
+    VBlk,
+    VBody,
+    WABlk,
+    WABody,
+)
+
+
+def create_content_page(
+    wa_body_count: int = 1,
+    wa_gender_var: list[str] | None = None,
+) -> ContentPage:
+    """
+    Helper function to create pages needed for each test.
+
+    Parameters
+    ----------
+    wa_body_count : int
+        How many WhatsApp message bodies to create on the content page.
+    wa_gender_var: list[str]
+        Variation restriction
+    """
+    title = "default page"
+    home_page = HomePage.objects.first()
+    main_menu = PageBuilder.build_cpi(home_page, "main-menu", "Main Menu")
+    parent = main_menu
+
+    if wa_gender_var is None:
+        wa_gender_var = []
+
+    bodies = [
+        MBody(title, [MBlk("Messenger 1"), MBlk("Messenger 2")]),
+        SBody(title, [SBlk("SMS 1"), SBlk("SMS 2")]),
+        UBody(title, [UBlk("USSD 1"), UBlk("USSD 2")]),
+        VBody(title, [VBlk("Viber 1"), VBlk("Viber 2")]),
+    ]
+
+    for i in range(wa_body_count):
+        wa_body = f"WhatsApp {i+1}"
+
+        variation_messages = [
+            VarMsg(f"WhatsApp Varied {var+1}", gender=wa_gender_var[var])
+            for var in range(len(wa_gender_var))
+        ]
+
+        bodies.append(
+            WABody(title, [WABlk(wa_body, variation_messages=variation_messages)])
+        )
+
+    content_page = PageBuilder.build_cp(
+        parent=parent,
+        slug=title.replace(" ", "-"),
+        title=title,
+        bodies=bodies,
+        tags=[],
+        quick_replies=[],
+        triggers=[],
+        publish=True,
+    )
+    return content_page
+
+
+@pytest.mark.django_db
+class TestHasNextMessage:
+    def test_next_messages_whatsapp(self) -> None:
+        """WhatsApp The next message after the last message is None.
+        Index of first message is 0, next message is then message 2 with index 1"""
```

Comment:
I don't think we've explicitly standardised on docstring formatting, but in most places we put the `"""` on their own line. Can we do the same in these tests?